libraries/AP_Motors/AP_MotorsMatrix_Scripting_Dynamic.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifdef ENABLE_SCRIPTING
+
+// This allows motor roll, pitch, yaw and throttle factors to be changed in flight, allowing vehicle geometry to be changed
+
+#include "AP_MotorsMatrix_Scripting_Dynamic.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define debug_print 0
+
+// add a motor and give its testing order
+bool AP_MotorsMatrix_Scripting_Dynamic::add_motor(int8_t motor_num, uint8_t testing_order)
```

Comment:
why signed motor_num?

Reply:
fixed, many of the other back ends have the same thing.